### PR TITLE
List the exposed metrics in the manual

### PR DESCRIPTION
**Pull request by spacecowboy**
The current metrics plugin docs don't list the actual metrics that are exposed.
This solution doesn't do much in the way of explaining what the metrics are, but the names are quite descriptive, and it is way better than not mentioning them at all.

### CODE REVIEW

**Comment by chrisvest** [comment]
You should add a note that the JVM metrics are just an example, and that the actual metrics will depend on the JVM configuration at runtime.

**Comment by davidegrohmann** [comment]
@nawroth what do you think about this change?

File: enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import static org.neo4j.metrics.MetricsSettings.csvFile;
+import static org.neo4j.metrics.MetricsSettings.csvPath;
+
+public class MetricsExposedListDocIT
+{
+
+    @Rule
+    public final TargetDirectory.TestDirectory folder = TargetDirectory.testDirForTest( getClass() );
+
+    GraphDatabaseService db;
```
Comment (davidegrohmann) [inline enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java]: missing `private`?

File: enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    private File outputFile;
+
+    @Before
+    public void setup() throws IOException
+    {
+        String dbPath = folder.directory( "data" ).getAbsolutePath();
+        outputFile = folder.file( "metrics.csv" );
+        db = new GraphDatabaseFactory().newEmbeddedDatabaseBuilder( dbPath ).
+                setConfig( GraphDatabaseSettings.allow_store_upgrade, Settings.TRUE ).
```
Comment (davidegrohmann) [inline enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java]: I don't see the need for this setting to be enabled.

File: enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            }
+        }
+
+        // Read the metrics we recorded
+        String[] headers;
+        try ( BufferedReader reader = new BufferedReader( new FileReader( outputFile ) ) )
+        {
+            // First verify that some metrics have been written
+            headers = reader.readLine().split( "," );
+            assertThat( headers[0], is( "timestamp" ) );
```
Comment (davidegrohmann) [inline enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java:99]: I am afraid this is not gonna be always true. Can't we search for that header instead?

Comment (spacecowboy) [inline enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java:99]: The test is more or less copy-pasted from `MetricsKernelExtensionFactoryIT`, and there we make exactly this assumption.

Comment (davidegrohmann) [inline enterprise/metrics/src/test/java/org/neo4j/metrics/MetricsExposedListDocIT.java:99]: :see_no_evil:

**Comment by benbc** [comment]
I don't think this approach is good enough for metrics that are explicitly coded in our code (rather than the JVM-generated ones that we don't control). We should just document the actual metrics and explain what they mean clearly, including the units used.

**Comment by davidegrohmann** [comment]
@benbc this is a first step, the next one is to add documentation for every entry we expose (exclude the JVM dependent ones) by annotating fields with `@Documented`

**Comment by benbc** [comment]
@davidegrohmann Okay, thanks.

**Comment by davidegrohmann** [comment]
@spacecowboy perhaps we should start adding `@Documented` to the fields defined by the various metrics and make sure the description in the annotation is used to generate the docs. WDYT?

**Comment by MishaDemianenko** [comment]
@spacecowboy @davidegrohmann that looks like good idea :), and update all recently introduced metrics definitions.
Also ping @lutovich

**Comment by nawroth** [comment]
It would be great if we can get `@Documented` in and use that. I'll mark this PR as not ready for merge meanwhile.

**Comment by thobe** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;ckkGf0CL&#x2F;1598-generate-documentation-for-metrics-exposed"> Generate Documentation for Metrics exposed</a>

**Comment by davidegrohmann** [comment]
Let's close this for now. Reopening it when ready.

**Pull request closed without merging**